### PR TITLE
Add ability to browse through data by most recent

### LINKED ISSUE #103: Add ability to browse through data by most recent
Display a list of submission sorted by the most recent for user to scroll through

### PR DESCRIPTION
**Pull request by hulloanson**
TODOs

- [x] Add submission list panel

- [x] "View all" button in panel and expand submission list to the entirety of the sidebar when clicked

- [x] Click submission to focus pin

- [x] Click pin to focus submission

- ~~[ ] Focus animation for submission list when pin is clicked~~

- ~~[ ] Focus animation for pin when submission is clicked~~

- [x] Filters should apply to list of reports (will also prevent click on filtered-out issue not zooming to pin)

- [x] put feed into its own tab

- [x] focus feed tab when user wants to open it in sidebar

### CODE REVIEW

**Comment by hulloanson** [comment]
@tallcoleman @sonalranjit It is ready for review.

One important change made that is not directly related to the goal of this PR: 

- Uses "hash" path (like how it's done with e.g. react-dom-router's HashRouter) to partially save UI state (e.g. useful when sharing a particular submission for example)

Therefore:

- There's a new class called HashRouter. PanelNav now instantiates it and put it into SharedState

- `PanelNav`, `Submissions` and `Map` uses HashRouter

  - `PanelNav` uses it to decide which tab to show

  - `Map` listens to param changes in hash url to zoom to a marker. Clicking the "focus in sidebar" link changes the hash url to the submission
  
  - `Submission` listens to param changes in hash url to scroll to and focus a submission.
  
Please try it out and see what you think!

Also I merged several times with `main`. Can @tallcoleman the new features in main work as expected?

**Comment by tallcoleman** [comment]
Thanks @hulloanson I'll give it a look through (and will check to make sure it works with the commits from main as well)

**Comment by hulloanson** [comment]
@tallcoleman ready to merge?

**Comment by hulloanson** [comment]
@tallcoleman Hash routing doc updated in README

File: bikespace_dashboard/js/components/hash_router.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+   */
+  onChange(cb) {
+    if (!this._callbacks.includes(cb)) {
+      this._callbacks.push(cb);
+    }
+  }
+
+  executeCallbacks() {
+    for (const cb of this._callbacks) {
+      setTimeout(() => cb(), 0);
```
Comment (tallcoleman) [inline bikespace_dashboard/js/components/hash_router.js:111]: More of a question of curiosity - what's the purpose of using `setTimeout` here? To make the execution async?

**Review by tallcoleman — APPROVED** [review]
Fantastic job, thank you @hulloanson ! This is such a cool feature and really pulls you in to exploring all the different issues reported on the map.

File: bikespace_dashboard/js/components/hash_router.js (hunk 1 above)
Comment (hulloanson) [reply bikespace_dashboard/js/components/hash_router.js:111]: Yes. Basically puts the call to the end of the job queue as a callback is potentially compute heavy (though it shouldn't be) and I want other UI code to execute first.

**Pull request merged**